<!-- 
    ROS navigation stack with velocity smoother and safety (reactive) controller
-->
<launch>

  
  <arg name="odom_frame_id"   default="robot_1/odom"/>
  <arg name="base_frame_id"   default="robot_1/base_footprint"/>
  <arg name="global_frame_id" default="map"/>
  <arg name="odom_topic" default="robot_1/odom" />
  <arg name="laser_topic" default="robot_1/scan" />
  <arg name="custom_param_file" default="$(find turtlebot_navigation1)/param/dummy.yaml"/>

  <node pkg="move_base" type="move_base" respawn="false" name="robot_1_move_base" output="screen">
    <rosparam file="$(find turtlebot_navigation1)/param/costmap_common_params.yaml" command="load" ns="global_costmap" />
    <rosparam file="$(find turtlebot_navigation1)/param/costmap_common_params.yaml" command="load" ns="local_costmap" />   
    <rosparam file="$(find turtlebot_navigation1)/param/local_costmap_params.yaml" command="load" />   
    <rosparam file="$(find turtlebot_navigation1)/param/global_costmap_params.yaml" command="load" />
    <rosparam file="$(find turtlebot_navigation1)/param/dwa_local_planner_params.yaml" command="load" />
    <rosparam file="$(find turtlebot_navigation1)/param/move_base_params.yaml" command="load" />
    <rosparam file="$(find turtlebot_navigation1)/param/global_planner_params.yaml" command="load" />
    <rosparam file="$(find turtlebot_navigation1)/param/navfn_global_planner_params.yaml" command="load" />
    <!-- external params file that could be loaded into the move_base namespace -->
    <rosparam file="$(arg custom_param_file)" command="load" />
    
    <!-- reset frame_id parameters using user input data -->
    <param name="global_costmap/global_frame" value="$(arg global_frame_id)"/>
    <param name="global_costmap/robot_base_frame" value="$(arg base_frame_id)"/>
    <param name="local_costmap/global_frame" value="$(arg odom_frame_id)"/>
    <param name="local_costmap/robot_base_frame" value="$(arg base_frame_id)"/>
    <param name="DWAPlannerROS/global_frame_id" value="$(arg odom_frame_id)"/>

    
    <remap from="move_base/goal" to="robot_1/move_base/goal"/>
    <remap from="move_base_simple/goal" to="robot_1/move_base_simple/goal"/>
    <remap from="cmd_vel" to="robot_1/navigation_velocity_smoother/raw_cmd_vel"/>
    <remap from="odom" to="$(arg odom_topic)"/>
    <remap from="scan" to="$(arg laser_topic)"/>
  </node>
</launch>
